zilencer/management/commands/populate_db.py

```diff
@@ -829,28 +829,29 @@ def assign_time_zone_by_delivery_email(delivery_email: str, new_time_zone: str)
             UserProfile.objects.filter(is_bot=False, realm=zulip_realm)
         )
 
-        # As we plan to change the default values for 'automatically_follow_topics_policy' and
-        # 'automatically_unmute_topics_in_muted_streams_policy' in the future, it will lead to
-        # skewing a lot of our tests, which now need to take into account extra events and database queries.
-        #
-        # We explicitly set the values for both settings to 'AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_NEVER'
-        # to make the tests independent of the default values.
-        #
-        # We have separate tests to verify events generated, database query counts,
-        # and other important details related to the above-mentioned settings.
-        for user in user_profiles:
-            do_change_user_setting(
-                user,
-                "automatically_follow_topics_policy",
-                UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_NEVER,
-                acting_user=None,
-            )
-            do_change_user_setting(
-                user,
-                "automatically_unmute_topics_in_muted_streams_policy",
-                UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_NEVER,
-                acting_user=None,
-            )
+        if options["test_suite"]:
+            # As we plan to change the default values for 'automatically_follow_topics_policy' and
+            # 'automatically_unmute_topics_in_muted_streams_policy' in the future, it will lead to
+            # skewing a lot of our tests, which now need to take into account extra events and database queries.
```

Comment:
> I don't love having a testsuite that differs from how production will actually run. Have we tried dropping this exception and seeing how many tests are actually affected, and how hard it would be to update those tests?

Yes, we went the other way. There were a lot of tests getting skewed, so we picked up this path. We had a [discussion ](https://chat.zulip.org/#narrow/stream/101-design/topic/automatically.20follow.20topics.20.2325914/near/1609293)around this on CZO

Reply:
> The commit message calls this "bug" and "incorrect behaviour" 

Changing it to NEVER for the development database too is the incorrect behavior. Our intention is to only change it for test-database.

Reply:
Our message-send code path tests are pretty extensive for all values of these settings, so I'm comfortable with having these values for the defaults -- changing this will mean that every test that needs to create a test message by sending one to a new topic will do the extra work to follow the topic, which is usually not valuable for the test.

Reply:
The argument that "tests will do extra work they don't need" feels like an insufficient argument to me.  Like, a large number of tests probably don't care about UserMessage rows for other users.  Should we mark every test user as soft-deactivated by default, except when explicitly needed, so we don't have the overhead of making those rows?  That would feels silly because it would leave the normal production operating regime _much_ less well-tested.

Specifically, the risk is that the "extra work" you call out _does something unexpected_ when interacting with other subsystems.  If the setting has some unexpected action at a distance when it's not `NEVER`, then we're not going to catch it because the only time we test with a non-`NEVER` value is in a very small number of tests while examining the things we _expect_ to change.  Sure, this is always somewhat a risk with testing the combinatorial exposition of configuration values, but the risk is heightened here by _the production-default-value being the configuration that is largely untested_.

Test diverging from production is a code smell which I think exposes us to unnecessary risk.

Reply:
I'm OK merging this as-is, with the tests set to `NEVER` -- we can hash out the testing approach on CZO and update if need be after this is merged.